src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -137,7 +146,9 @@ private class UncommonProperties
             internal string lazyDefaultMemberName;
             internal NamedTypeSymbol lazyComImportCoClassType = ErrorTypeSymbol.UnknownResultType;
             internal ThreeState lazyHasEmbeddedAttribute = ThreeState.Unknown;
+            internal NativeIntegerTypeSymbol lazyNativeInt;
```

Comment:
>internal NativeIntegerTypeSymbol lazyNativeInt; [](start = 12, length = 47)

I think we can cache this on the assembly symbol. It feels wasteful to spend this memory for every "uncommon" type symbol just so we could use this slot on two types.

Reply:
See MetadataOrSourceAssemblySymbol._lazySpecialTypes

---
In reply to: [382297303](https://github.com/dotnet/roslyn/pull/41728#discussion_r382297303) [](ancestors = 382297303)

Reply:
We can simply add two more slots at the end of the array

---
In reply to: [382299283](https://github.com/dotnet/roslyn/pull/41728#discussion_r382299283) [](ancestors = 382299283,382297303)